pkg/kvstore/etcd.go

```diff
@@ -574,17 +574,17 @@ func connectEtcdClient(config *client.Config, cfgPath string, errChan chan error
 		ec.getLogger().Debugf("Session received")
 		close(ec.firstSession)
 
-		if err := ec.checkMinVersion(); err != nil {
+		if err := ec.checkMinVersion(ctx); err != nil {
 			errChan <- fmt.Errorf("unable to validate etcd version: %s", err)
 		}
 	}()
 
-	go ec.statusChecker(context.TODO())
+	go ec.statusChecker(ctx)
```

Comment:
We need to be careful with these contexts. In this case it's not sane passing a context to this function since it's a go routine running in the background. If `ctx` has a timeout of 30 seconds, this same go routine will be closed after 30 seconds. same with 1)